core/src/services/swift/backend.rs

```diff
@@ -0,0 +1,335 @@
+            Some(account) => account,
+            None => {
+                return Err(Error::new(
+                    ErrorKind::ConfigInvalid,
+                    "missing account name for Swift",
+                ));
+            }
+        };
+        debug!("backend use account: {}", &account);
+
+        let container = match self.container.take() {
+            Some(container) => container,
+            None => {
+                return Err(Error::new(
+                    ErrorKind::ConfigInvalid,
+                    "missing container for Swift",
+                ));
+            }
+        };
+        debug!("backend use container: {}", &container);
+
+        let token = match self.token.take() {
+            Some(token) => token,
+            None => String::new(),
+        };
+
+        let client = HttpClient::new()?;
+
+        debug!("backend build finished: {:?}", &self);
+        Ok(SwiftBackend {
+            core: Arc::new(SwiftCore {
+                root,
+                endpoint,
+                account,
+                container,
+                token,
+                client,
+            }),
+        })
+    }
+}
+
+/// Backend for Swift service
+#[derive(Debug, Clone)]
+pub struct SwiftBackend {
+    core: Arc<SwiftCore>,
+}
+
+#[async_trait]
+impl Accessor for SwiftBackend {
+    type Reader = IncomingAsyncBody;
+    type BlockingReader = ();
+    type Writer = oio::OneShotWriter<SwiftWriter>;
+    type BlockingWriter = ();
+    type Pager = SwiftPager;
+    type BlockingPager = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::Swift)
+            .set_root(&self.core.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                read: true,
+                read_can_next: true,
+                read_with_range: true,
+
+                write: true,
+                create_dir: true,
+                delete: true,
+                rename: true,
+
+                list: true,
+                list_with_delimiter_slash: true,
+
+                ..Default::default()
+            });
+        am
+    }
+
+    async fn create_dir(&self, path: &str, _args: OpCreateDir) -> Result<RpCreateDir> {
+        let resp = self
+            .core
+            .swift_create_object(path, AsyncBody::Empty)
+            .await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::CREATED | StatusCode::OK => {
+                resp.into_body().consume().await?;
+                Ok(RpCreateDir::default())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let range = args.range();
+        let resp = self.core.swift_read(path, range).await?;
+
+        match resp.status() {
+            StatusCode::OK | StatusCode::PARTIAL_CONTENT => {
```

Comment:
We should handle `Range Not Satisfiable (416)` correctly.

Reply:
Done